Wikipedia/Code/TalkPagesController.swift

```diff
@@ -147,10 +131,9 @@ class TalkPageController {
         }
         
         taskGroup.waitInBackground {
-            
-            self.localHandler.dataStore.viewContext.perform {
+            moc.perform {
                 if talkPageDoesNotExist {
-                    if let newLocalTalkPage = self.localHandler.createEmptyTalkPage(with: taskURL, languageCode: self.languageCode, displayTitle: self.displayTitle) {
+                    if let newLocalTalkPage = moc.createEmptyTalkPage(with: taskURL, languageCode: self.languageCode, displayTitle: self.displayTitle) {
                         completion?(.success(newLocalTalkPage))
```

Comment:
Maybe these completion calls within the waitInBackground should be wrapped up in a Dispatch.main.async?

Reply:
looks like `waitInBackgroundWithCompletion` notifies on the main queue by default - https://github.com/wikimedia/wikipedia-ios/blob/dad1b8ba65bbdb4fe29c8b1620a9ef4052078f9b/Wikipedia/Code/WMFTaskGroup.m#L48

Reply:
missed that this was inside a `waitInBackground` and `perform`, disregard 👆